client/client.go

```diff
@@ -243,80 +214,199 @@ func (c *Client) update(latestRoot bool) (data.TargetFiles, error) {
 		}
 	}
 
-	// Save the snapshot.json now it has been processed successfully
-	if err := c.local.SetMeta("snapshot.json", snapshotJSON); err != nil {
-		return nil, err
-	}
-
 	return updatedTargets, nil
 }
 
-func (c *Client) updateWithLatestRoot(m *data.SnapshotFileMeta) (data.TargetFiles, error) {
-	var rootJSON json.RawMessage
-	var err error
-	if m == nil {
-		rootJSON, err = c.downloadMetaUnsafe("root.json", defaultRootDownloadLimit)
-	} else {
-		rootJSON, err = c.downloadMetaFromSnapshot("root.json", *m)
+func (c *Client) updateRoots() error {
+	// https://theupdateframework.github.io/specification/v1.0.19/index.html#load-trusted-root
+	// Temporarily deactivate expiration check. This is consistent with 5.3.6:
+	// Note that the expiration of the trusted root metadata file does not
+	// matter, because we will attempt to update it in the next step.
+	e := verify.IsExpired
+	defer func() { verify.IsExpired = e }()
+	verify.IsExpired = func(t time.Time) bool { return false }
```

Comment:
You added `UnmarshalIgnoreExpired` but it doesn't seem to be used anymore. I think you should remove the verify.IsExpired override from non testing code and use the IgnoreExpired func.
Exemple of side effect of updating a global variable, if two different clients are run concurrently on the same process, there's a race condition on verify.IsExpired

Adding a test and running `go test -race` in the CI would catch it
```
func TestRace(t *testing.T) {
        for i := 0; i < 2; i++ {
                go func() {
                        c := NewClient(MemoryLocalStore(), newFakeRemoteStore())
                        c.Update()
                }()
        }
}
==================
WARNING: DATA RACE
Read at 0x000001aa7330 by goroutine 10:
  github.com/theupdateframework/go-tuf/client.(*Client).updateRoots()
      /Users/raphael.gavache/go/src/github.com/hosseinsia/go-tuf/client/client.go:225 +0x64
  github.com/theupdateframework/go-tuf/client.(*Client).Update()
      /Users/raphael.gavache/go/src/github.com/hosseinsia/go-tuf/client/client.go:155 +0x55
  github.com/theupdateframework/go-tuf/client.TestRace.func1()
      /Users/raphael.gavache/go/src/github.com/hosseinsia/go-tuf/client/client_test.go:1158 +0x164

Previous write at 0x000001aa7330 by goroutine 8:
  github.com/theupdateframework/go-tuf/client.(*Client).updateRoots()
      /Users/raphael.gavache/go/src/github.com/hosseinsia/go-tuf/client/client.go:227 +0xbe
  github.com/theupdateframework/go-tuf/client.(*Client).Update()
      /Users/raphael.gavache/go/src/github.com/hosseinsia/go-tuf/client/client.go:155 +0x55
  github.com/theupdateframework/go-tuf/client.TestRace.func1()
      /Users/raphael.gavache/go/src/github.com/hosseinsia/go-tuf/client/client_test.go:1158 +0x164
```

Reply:
fixed.